hummingbot/connector/exchange/bittrex/bittrex_api_order_book_data_source.py

```diff
@@ -1,244 +1,178 @@
-                diff_timestamp = diff["nonce"]
-                diff_msg: OrderBookMessage = BittrexOrderBook.diff_message_from_exchange(
-                    diff["results"], diff_timestamp
-                )
-                output.put_nowait(diff_msg)
-            except Exception:
-                self.logger().error("Unexpected error when listening on socket stream.", exc_info=True)
-
+        self._connector = connector
+        self._trade_messages_queue_key = CONSTANTS.TRADE_EVENT_KEY
+        self._diff_messages_queue_key = CONSTANTS.DIFF_EVENT_KEY
+        self._api_factory = api_factory
+        self.hub = None
+
+    async def _connected_websocket_assistant(self):
+        ws_connection = signalr_aio.Connection(CONSTANTS.BITTREX_WS_URL, session=None)
+        self.hub = ws_connection.register_hub("c3")
+        return ws_connection
+
+    async def get_last_traded_prices(self,
+                                     trading_pairs: List[str],
+                                     domain: Optional[str] = None) -> Dict[str, float]:
+        return await self._connector.get_last_traded_prices(trading_pairs=trading_pairs)
+
+    async def _request_order_book_snapshot(self, trading_pair: str) -> Dict[str, Any]:
+        exchange_symbol = await self._connector.exchange_symbol_associated_to_pair(trading_pair=trading_pair)
+        params = {
+            "marketSymbol": exchange_symbol,
+        }
+        rest_assistant = await self._api_factory.get_rest_assistant()
+        data = await rest_assistant.execute_request(
+            url=web_utils.public_rest_url(path_url=CONSTANTS.ORDERBOOK_SNAPSHOT_URL.format(exchange_symbol)),
+            params=params,
+            method=RESTMethod.GET,
+            throttler_limit_id=CONSTANTS.ORDERBOOK_SNAPSHOT_LIMIT_ID,
+        )
+        return data
+
+    async def _order_book_snapshot(self, trading_pair: str) -> OrderBookMessage:
+        snapshot: Dict[str, Any] = await self._request_order_book_snapshot(trading_pair)
+        snapshot_timestamp: float = time.time()
+        snapshot_msg: OrderBookMessage = self.snapshot_message_from_exchange(
+            snapshot,
+            snapshot_timestamp,
+            metadata={"trading_pair": trading_pair}
+        )
+        return snapshot_msg
+
+    # TODO Move utility functions to one single file after verification
     async def _checked_socket_stream(self, connection: signalr_aio.Connection) -> AsyncIterable[str]:
         try:
             while True:
-                async with timeout(MESSAGE_TIMEOUT):  # Timeouts if not receiving any messages for 10 seconds(ping)
+                async with timeout(CONSTANTS.MESSAGE_TIMEOUT):  # Timeouts if not receiving any messages for 10 seconds(ping)
                     msg = await connection.msg_queue.get()
                     yield msg
         except asyncio.TimeoutError:
             self.logger().warning("Message queue get() timed out. Going to reconnect...")
```

Comment:
Doesnt seem like we are doing anything with this timeout other than logging it. If there's no reason for this timeout, lets remove it

Reply:
This function is no longer required and is removed.